### ISSUE TITLE
Card Layout Component

### ISSUE THREAD

**Issue by droshev** [SAP/fundamental-ngx#5944]
#### Is this a bug, enhancement, or feature request?
feature request

#### Briefly describe your proposal.

- (1) display cards with different dimensions in a dashboard way
- (2) Display/Edit mode toggle(maybe more action buttons)
- (1) provide grid component - different options? - check what luigi offers - we can start with those and add more. _Follow with designers and maybe support at least Fiori/BTP grids_ Layout per form factor (and how to store that)
- (1) nodes[] - list of nested compound views(which come with metadata for where to be rendered - row/column)
- (2) allow changing the cards positions(drag and drop) and offer to export the new layout(for future storing); allow importing the layout when the grid is loaded
- (1) fundamental can offer rendering the cards wrapper, including the header section which comes from the metadata, the application provides the content in the body of the card. Application can come with full card(including the header). In such cases fundamental won't render anything additionally. (Offer both options)
- (1) export and publish it as a WC(offer slots in that cases) - depends on Phase 0
- (2)Edit mode: add/remove cards from the layout

Phase 0: POC of which technology to use for providing performant WC (stencils, angular element including webpack module federation, svelte, ). Another PoC needs to be done how to position the cards (column layout or css grid or flex box) => this can define the position spec. check the different card examples from the design

Phase 0.1 - Web components technologies(e.g. angular elements, or others - open to bring more)
Phase 0.2 - how to position the cards (column layout or css grid or flex box) => this can define the position spec. check the different card examples from the design
Phase 1: deliver display mode, the layout/order will come as input.(x/y starting point and x/y span)
Phase 2: Edit mode ( there are more open questions atm)
Phase 3: Adaptive Cards - needs a separate discussion

Links:
- [how to share libs](https://www.angulararchitects.io/aktuelles/the-microfrontend-revolution-part-2-module-federation-with-angular/)
- [angular example](https://tiberiuzuld.github.io/angular-gridster2/drag)
- [example](https://github.com/gridstack/gridstack.js)
- look for other existing libs

Out of scope:
- card resizing
- (TBD) add card button will bring a new dialog with list of card types. once the user select a type then another dialog will pop where the user can configure the card data
- (2)Edit mode: drag and drop, card resize(not all the cards could be resizable)
- (3) explore using adaptive cards(ms) and see how they can be used

POC
- @N1XUS - to evaluate the grid/drag-n-drop libraries out there; sync with Tino, Colin to understand any constrains 
Aug 3 - we met with Philipp, Tino, Colin and Denis. We reviewed what Denis did using gridster.

- Create a quick PoC - Dashboard angular element and include different angular elements inside in diff slots(the slot elements should be from diff libraries e.g. one angular element, ui5 wc)

**Comment by tjbutz** [comment]
Another example / reuse component: 

https://muuri.dev/

**Comment by tjbutz** [comment]
Talked to the UX lead and he likes the behavior of https://gridstackjs.com/#demo

**Comment by N1XUS** [comment]
@droshev do we have other items that should be covered by this ticker, or it can be closed?
